### ISSUE TITLE
Cypress trying to write to /root

### ISSUE THREAD

**Issue by mofojed** [cypress-io/cypress#8283]
### Current behavior:
Cypress is failing when building in Jenkins with the latest 4.12.1, as it's trying to write to `/root/.cache/Cypress` in it's postinstall script:
```
> cypress@4.12.1 postinstall /home/jenkins/agent/workspace/iris/web/client-ui/build/js/node_modules/cypress
> node index.js --exec install

Cypress cannot write to the cache directory due to file permissions

See discussion and possible solutions at
https://github.com/cypress-io/cypress/issues/1281

----------

Failed to access /root/.cache/Cypress:

EACCES: permission denied, mkdir '/root/.cache/Cypress'
```
According to the link, this issue should have been fixed in a previous version, but looks like it's back. Reverting to my previous version (4.3.0) fixed the issue.

### Desired behavior:

Cypress should not try to write to root, and if can't, it should warn and work without caching. Or there should be an option to configure the cache, or it shouldn't run automatically in the postinstall script.

### Test code to reproduce
Run `npm install` for a package using Cypress from a user that doesn't have root access.

### Versions
Platform: linux (CentOS - 7.7.1908)
Cypress Version: 4.12.1

**Comment by jennifer-shehane** [comment]
Duplicate of #7640

**Comment by mofojed** [comment]
@jennifer-shehane I see the comment on #7640 about setting correct permissions - what are the correct permissions to set? And why is there a different in behaviour between 4.3.0 and 4.12.1? I wasn't getting this error on install with 4.3.0 previously, and I don't see anything in the changelogs related to updating permissions because of migrating to a new version.

**Comment by capetown** [comment]
I abandoned docker based Cypress integration with Jenkins because of this issue. I followed the mapped non-root user example exactly. As a new Cypress user, I assumed it was something I missed. Using version 4.12.0 of the bundled docker+cypress image.

**Comment by Dziejo93** [comment]
@mofojed Hey, had something like this however on ubuntu 20.04. Cypress kept crashing throwing different errors based on permissions, or even no cypress installation present. What I did that worked for me was to remove contents of `~/.cache/Cypress` and `~./config/Cypress` and then run installation from node_modules.

**Comment by amav96** [comment]
No hay solucion?

**Comment by nitsan** [comment]
I fixed it with my own docker:

```
FROM cypress/browsers:node14.17.0-chrome91-ff89

RUN npm i -g npm@8

RUN chown -R node:node /root

USER node

RUN mkdir /root/.cache
```

Then push it you your ECR and use it

**Comment by okgolove** [comment]
Since 7 version NPM removed some parameters like user and changed behaviour for NPM running from root.

https://github.com/npm/cli/blob/latest/changelogs/CHANGELOG-7.md#all-lifecycle-scripts
>When npm is run as root, scripts are always run with the effective uid and gid of the working directory owner.

Check a directory you run npm from permissions:
```
# Linux
stat -c 'userid - %u, groupid - %g' $PWD

#MacOS
stat -f 'userid - %u, groupid - %g' $PWD
```

In case of it is not 0 (root) and you run `npm i` or `npm ci` from root cypress won't be able to create directory `/root/.cache/Cypress`

In this particular case Jenkins clones repository with its uid and gid (if it runs in Docker it is 1000:1000). So, when you try run `npm install` npm tries to run setup scripts with uid=1000:gid=1000

**Comment by webia1** [comment]
How can I configure another cache directory in a Jenkins file for cypress? Are there other possibilities? 
Thanks for feedbacks, kind rgds

**Comment by DeividVeloso** [comment]
Same here

**Comment by JUPUDI-VAMSI-KALYAN** [comment]
> Same here

Use the following Environment Variables issue will be resolved
```
ENV HOME /app/.cache
ENV CYPRESS_CACHE_FOLDER /app/.cache
```

**Comment by DeividVeloso** [comment]
This is my error @JUPUDI-VAMSI-KALYAN 

```
npm ERR! code 1
npm ERR! path /app/node_modules/cypress
npm ERR! command failed
npm ERR! command sh -c node index.js --exec install
npm ERR! Cypress cannot write to the cache directory due to file permissions
npm ERR! 
npm ERR! See discussion and possible solutions at
npm ERR! https://github.com/cypress-io/cypress/issues/1281
npm ERR! 
npm ERR! ----------
npm ERR! 
npm ERR! Failed to access /root/.cache/Cypress:
npm ERR! 
npm ERR! EACCES: permission denied, mkdir '/root/.cache/Cypress'
```

**Comment by JUPUDI-VAMSI-KALYAN** [comment]
@DeividVeloso  You have to give permissions check my code

`FROM cypress/base:16.14.2
RUN mkdir /app
WORKDIR /app
COPY . /app 
ENV HOME /app/.cache
ENV CYPRESS_CACHE_FOLDER /app/.cache
RUN npm install --unsafe-perm=true --allow-root
RUN $(npm bin)/cypress verify
ENTRYPOINT []
RUN groupadd -r user && useradd -r -g user user
RUN chown -R user:user /app
RUN chmod 755 /app
USER user
CMD node main.js`

**Comment by DeividVeloso** [comment]
@JUPUDI-VAMSI-KALYAN 

```
WORKDIR /app

ENV HOME /app/.cache
ENV CYPRESS_CACHE_FOLDER /app/.cache

# RUN addgroup user && adduser -g user root
RUN chown -R root:root /app
RUN chmod 755 /app

# https://github.com/cypress-io/cypress/issues/1281
# EACCES: permission denied, mkdir '/root/.cache/Cypress'
RUN mkdir /app/.cache  
```

I will test this

**Comment by marcosaguilera-glober** [comment]
I was getting same issue:
**# EACCES: permission denied, mkdir '/root/.cache/Cypress'**

I've fixed like this:
1. In my Dockerfile
```
#Cypress version using
FROM cypress/included:10.6.0

# Installing sudo for superower installation required
# into entrypoint.sh
RUN apt-get install sudo

# Directories actions
WORKDIR /home/pwsh/tempVolume

ENTRYPOINT ["/bin/sh", "./entrypoint.sh"]
```

2. In my entrypoint.sh script file
```
#!/bin/sh
echo "I'm into the sh file"

## Install packages defined into package.json > dependencies node
##npm install
sudo npm install --unsafe-perm=true --allow-root

## Run cypress
cypress run
```
I messed up a time with this issue, and I tried with permissions `chown node:node`, also some `chmod -R 777`, but nothing worked for me, so I installed `sudo apt-get` into Linux distro.

Hope this help anyone.

**Comment by JUPUDI-VAMSI-KALYAN** [comment]
@DeividVeloso @marcosaguilera-glober 
`FROM cypress/browsers:node16.14.2-slim-chrome100-ff99-edge

RUN mkdir /app

WORKDIR /app

COPY . /app

ENV HOME /app/.cache

ENV CYPRESS_CACHE_FOLDER /app/.cache

RUN npm install

RUN $(npm bin)/cypress verify

ENTRYPOINT []

RUN chgrp -R 0 /app && chmod -R g+rwX /app

USER 1001

RUN npx browserslist@latest --update-db

CMD node scheduler.js`

By Specifying Home and CYPRESS_CACHE_FOLDER environment variables. this issue will get resolved.Check the above docker code. Thankyou
